wallet/wallet.go

```diff
@@ -2712,20 +2727,38 @@ func (w *Wallet) PublishTransaction(tx *wire.MsgTx) error {
 	// we'll write this tx to disk as an unconfirmed transaction. This way,
 	// upon restarts, we'll always rebroadcast it, and also add it to our
 	// set of records.
-	rec, err := wtxmgr.NewTxRecordFromMsgTx(tx, time.Now())
+	txRec, err := wtxmgr.NewTxRecordFromMsgTx(tx, time.Now())
 	if err != nil {
 		return err
 	}
 	err = walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
 		txmgrNs := tx.ReadWriteBucket(wtxmgrNamespaceKey)
-		return w.TxStore.InsertTx(txmgrNs, rec, nil)
+		return w.TxStore.InsertTx(txmgrNs, txRec, nil)
 	})
 	if err != nil {
 		return err
 	}
 
 	_, err = server.SendRawTransaction(tx, false)
-	return err
+	if err != nil {
+		// If the transaction was rejected, then we'll remove it from
```

Comment:
I believe this would mean that calling `PublishTransaction()` twice with the same transaction removes it from the txstore the second time. This is only a nuisance since it's only removed if it's unmined, which means it'll be re-detected once mined, but it may throw off balance/tx info until then.

Reply:
Hmm, yeah I think you're correct. 

Upon a second pass, the additional clause I added to tx retransmission is too encompassing, so things like "already in mempool" would results in transactions being deleted. 

Will modify to target precise errors.